dali/core/mm/default_resource_test.cc

```diff
@@ -374,54 +388,88 @@ TEST(MMDefaultResource, DISABLED_ReleaseUnusedMaxMem) {
   ReleaseUnusedTestImpl();
 }
 
-static void PreallocateTestImpl() {
+TEST(MMDefaultResource, PreallocatePinnedMemory) {
+  mm::ReleaseUnusedMemory();  // release any unused memory to check that we're really preallocating
+
+  auto *res = mm::GetDefaultResource<mm::memory_kind::pinned>();
+  auto *pool = GetPoolInterface(res);
+  if (!pool)
+    GTEST_SKIP() << "No memory pool in use - cannot test pool preallocation.";
+
+  size_t size = 64_uz << 20;  // 64 MiB
+  size_t alignment = alignof(std::max_align_t);  // default alignment
+  // Try to get increasing amount of memory from the pool until we can't
+  for (;; size <<= 1) {
+    void *mem = pool->try_allocate_from_free(size, alignment);
+    if (!mem)
+      break;
+    res->deallocate(mem, size, alignment);
+  }
+
+  std::cout << "Preallocating " << (size >> 20) << " MiB of pinned memory" << std::endl;
+
+  try {
+    // Try to preallocate the pool so we're able to get the requested amount
+    mm::PreallocatePinnedMemory(size);
+  } catch (const std::bad_alloc &) {
+    GTEST_SKIP() << "Not enough memory to test pool preallocation.";
+  }
+
+  void *mem = pool->try_allocate_from_free(size, alignment);
+  EXPECT_NE(mem, nullptr) << "Preallocation succeeded, so we should be able to get the "
+                             "requested amount of memory from the pool.";
+
+  res->deallocate(mem, size, alignment);
+
+  mm::ReleaseUnusedMemory();
+}
+
+TEST(MMDefaultResource, PreallocateDeviceMemory) {
```

Comment:
```suggestion
TEST(MMDefaultResource, MultiGPUPreallocateDeviceMemory) {
```
So we run it as a part of `TL0_multigpu`

Reply:
It's not _that_ important. I wanted some extra coverage (and indeed it helped uncover some old bugs), but the test has its merit also when there's only one GPU.